zoo_calrissian_runner/__init__.py

```diff
@@ -221,19 +223,38 @@ def get_input_value(self, key):
     def get_processing_parameters(self):
         """Returns a list with the input parameters keys"""
         res={}
+        hasVal=False
         for key, value in self.inputs.items():
             if "dataType" in value:
-                match value["dataType"]:
-                    case w if w in ["double","float"]:
+                if isinstance(value["dataType"],list):
+                    # How should we pass array for an input?
+                    import json
+                    res[key]=value["value"]
+                else:
+                    if value["dataType"] in ["double","float"]:
                         res[key]=float(value["value"])
-                    case "integer":
+                    elif value["dataType"] == "integer":
                         res[key]=int(value["value"])
-                    case "boolean":
+                    elif value["dataType"] == "boolean":
                         res[key]=bool(value["value"])
-                    case _:
+                    else:
                         res[key]=value["value"]
```

Comment:
Any particular reason to switch from case to if/else? and any logical reason why list is separated from the other if/elif chain?

Reply:
good point, copied from parent repo, I will update this to be case statement.
I expect `list` is handled separately as it means each item in the array could be a different datatype, so will leave this as is

Reply:
This causes an issue due to the Python version used by the ZOO Project build, so for now I have set this back to if...else